src/TBCapabilityFactory.js

```diff
@@ -57,12 +54,19 @@ export class CapabilityFactory {
 		let capabilityResult = Object.assign( {}, this.defaultCapabilities ) ;
 
 		if ( dimensions.has( Dimension.DEVICE ) ) {
-			const deviceName = dimensions.get( Dimension.DEVICE );
-			if ( !DEVICE_NAMES.has( deviceName ) ) {
-				throw new Error( `Unsupported device: ${deviceName}` )
+			const requestedTestPlatform = dimensions.get( Dimension.DEVICE );
+			if ( !DEVICE_NAMES.has( requestedTestPlatform ) ) {
+				throw new Error( `Unsupported device: ${requestedTestPlatform}` )
 			}
-			capabilityResult = { ...DEVICE_NAMES.get( deviceName ) };
-			capabilityResult.orientation = dimensions.has( Dimension.ORIENTATION ) ? dimensions.get( Dimension.ORIENTATION ).toUpperCase() : 'PORTRAIT';
+			capabilityResult = { ...DEVICE_NAMES.get( requestedTestPlatform ) };
+
+			capabilityResult['appium:deviceName'] = DEVICE_NAMES.get( requestedTestPlatform ).deviceName;
+			delete capabilityResult['deviceName'];
```

Comment:
You could edit DEVICE_NAMES directly and add the `appium:` prefix there. I've searched the code of banner-screenshots and shutterbug and found no other usage of `deviceName` than in the capability setup. And since `TBCapabilityFactory` is specific to testingbot anyway, we might as well have testingbot-specific configuration in our defaults

Reply:
good idea.. I was way too stuck inside the problem to see that possibility